info-compiler/src/main/java/com/google/sps/webcrawler/WebCrawler.java

```diff
@@ -155,80 +160,118 @@ public void compileNewsArticle(String candidateName, String candidateId) throws
+      for (DateTimeFormatter potentialDateTimeFormatter : DATE_TIME_FORMATTERS) {
+        try {
+          return Date.from(Instant.from(potentialDateTimeFormatter.parse(date)));
+        } catch (IllegalArgumentException | DateTimeException e) {}
+      }
     }
-    return urls;
+    return null;
   }
 
   /**
    * Checks robots.txt for permission to web-scrape, scrapes webpage if permitted and extracts
-   * textual content. Returns an empty {@code Optional<NewsArticle>} in the event of an exception.
+   * textual content to put into {@code newsArticle}. Sets "content" to empty in the event of an
+   * exception.
    */
-  public Optional<NewsArticle> scrapeAndExtractFromHtml(URL url) {
+  public void scrapeAndExtractFromHtml(NewsArticle newsArticle) {
     try {
+      URL url = new URL(newsArticle.getUrl());
       URL robotsUrl = new URL(url.getProtocol(), url.getHost(), "/robots.txt");
       InputStream robotsTxtStream = setTimeoutAndOpenStream(robotsUrl);
       RobotsTxt robotsTxt = RobotsTxt.read(robotsTxtStream);
       robotsTxtStream.close();
       String webpagePath = url.getPath();
       Grant grant = robotsTxt.ask("*", webpagePath);
-      return politelyScrapeAndExtractFromHtml(grant, robotsUrl, url);
+      politelyScrapeAndExtractFromHtml(grant, robotsUrl, newsArticle);
     } catch (Exception e) {
       System.out.println("[ERROR] Error occured in scrapeAndExtractHtml(): " + e);
-      return Optional.empty();
+      newsArticle.setContent("");
     }
   }
 
   /**
    * Checks robots.txt for permission to web-scrape, scrapes webpage if permitted and extracts
-   * textual content. Returns an empty {@code Optional<NewsArticle>} in the event of an exception.
+   * textual content to put into {@code newsArticle}. Sets "content" to empty in the event of an
+   * exception.
    */
-  Optional<NewsArticle> politelyScrapeAndExtractFromHtml(Grant grant, URL robotsUrl, URL url) {
+  void politelyScrapeAndExtractFromHtml(Grant grant, URL robotsUrl,
+      NewsArticle newsArticle) {
     try {
       // Check permission to access and respect the required crawl delay.
       if (grant == null || grant.hasAccess()) {
         if (grant != null
             && grant.getCrawlDelay() != null
             && !waitForAndSetCrawlDelay(grant, robotsUrl.toString())) {
-          return Optional.empty();
+          newsArticle.setContent("");
+          return;
         }
-        InputStream webpageStream = setTimeoutAndOpenStream(url);
-        Optional<NewsArticle> potentialNewsArticle =
-            newsContentExtractor.extractContentFromHtml(webpageStream, url.toString());
+        InputStream webpageStream = setTimeoutAndOpenStream(new URL(newsArticle.getUrl()));
+        newsContentExtractor.extractContentFromHtml(webpageStream, newsArticle);
         webpageStream.close();
-        return potentialNewsArticle;
       } else {
-        return Optional.empty();
+        newsArticle.setContent("");
       }
```

Comment:
Can you clean up these if/else statements to just be a `if (something) { webpagestream stuff } else { set contant "" }`?

Reply:
Done!